src/main/java/de/konfidas/ttc/reporting/Report.java

```diff
@@ -0,0 +1,66 @@
+package de.konfidas.ttc.reporting;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Generic Class for reporting. The class represents a tree structure of elements
+ * that make up a report
+ * @param <T> The type of the data to be reported
+ */
+public class Report<T> {
```

Comment:
You should never implement basic datatypes yourself (like lists, trees, etc.). 

this is a simple tree with children and a generic datatype. there has to be some class for this in the java api to use. there is no need to do this on your own.

Reply:
I did not find any implementation in the Java framework for a tree (surprised me as well). Do you have a specific class in mind?

Reply:
I have to admit i do not. and a quick search did not bring up something useful.